pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -2515,6 +2522,57 @@ func (vs *vSphereVMProvider) vmCreateGenConfigSpec(
 	return nil
 }
 
+func (vs *vSphereVMProvider) vmCreateGenConfigSpecAffinityPolicy(
```

Comment:
I don't think this needs to be a provider method. I was kind of expecting something in the `virtualmachine` package along side `CreateConfigSpec()` (or just do it as a part of that function) so we can more easily test that the expected ConfigSpec is being created.

Reply:
I started with putting it in the virtualmachine package, but it seems we have a pattern of taking the generated ConfigSpec and modifying it to add other things. 
Like, these methods:
https://github.com/vmware-tanzu/vm-operator/blob/main/pkg/providers/vsphere/vmprovider_vm.go#L2503-L2513